gnomad_qc/v3/variant_qc/random_forest.py

```diff
@@ -0,0 +1,475 @@
+
+        logger.info("Adding run to RF run list")
+        rf_runs[model_id] = get_run_data(
+            input_args={
+                "transmitted_singletons": None if args.vqsr_training else not args.no_transmitted_singletons,
+                "adj": args.adj,
+                "vqsr_training": args.vqsr_training,
+                "filter_centromere_telomere": args.filter_centromere_telomere,
+            },
+            test_intervals=args.test_intervals,
+            features_importance=hl.eval(ht.features_importance),
+            test_results=hl.eval(ht.test_results),
+        )
+
+        with hl.hadoop_open(rf_run_path(), "w") as f:
+            json.dump(rf_runs, f)
+
+        logger.info("Saving RF model")
+        save_model(
+            rf_model, get_rf_model_path(model_id=model_id), overwrite=args.overwrite,
+        )
+
+    else:
+        model_id = args.model_id
+
+    if args.apply_rf:
+        logger.info(f"Applying RF model {model_id}...")
+        rf_model = load_model(get_rf_model_path(model_id=model_id))
+        ht = get_rf_training(model_id=model_id).ht()
+        features = hl.eval(ht.features)
+        ht = apply_rf_model(ht, rf_model, features, label=LABEL_COL)
+
+        logger.info("Finished applying RF model")
+        ht = ht.annotate_globals(rf_model_id=model_id)
+        ht = ht.checkpoint(
+            get_rf_result(model_id=model_id).path, overwrite=args.overwrite,
+        )
+
+        ht_summary = ht.group_by(
+            "tp", "fp", TRAIN_COL, LABEL_COL, PREDICTION_COL
+        ).aggregate(n=hl.agg.count())
+        ht_summary.show(n=20)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--slack_channel", help="Slack channel to post results and notifications to."
+    )
+    parser.add_argument(
+        "--overwrite",
+        help="Overwrite all data from this subset (default: False)",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--model_id",
+        help="Model ID. Created by --train_rf and only needed for --apply_rf without running --train_rf",
+        required=False,
+    )
+
+    actions = parser.add_argument_group("Actions")
+    actions.add_argument(
+        "--list_rf_runs",
+        help="Lists all previous RF runs, along with their model ID, parameters and testing results.",
+        action="store_true",
+    )
+    actions.add_argument(
+        "--annotate_for_rf",
+        help="Creates an annotated ht with features for RF",
+        action="store_true",
+    )
+    actions.add_argument("--train_rf", help="Trains RF model", action="store_true")
+    actions.add_argument(
+        "--apply_rf", help="Applies RF model to the data", action="store_true"
+    )
+    actions.add_argument("--finalize", help="Write final RF model", action="store_true")
+
+    annotate_params = parser.add_argument_group("Annotate features parameters")
+    annotate_params.add_argument(
+        "--impute_features_no_variant_type",
```

Comment:
See previous comment -- feature imputation doesn't happen at all if this argument is set to `False`